src/vs/code/browser/workbench/workbench.ts

```diff
@@ -55,6 +55,10 @@ type Writeable<T> = { -readonly [P in keyof T]: T[P] };
 	config.commands = [
 		// Used to refresh the page from the extension when a new version of the IDE is known to exist.
 		{ id: 'membrane.refreshPage', handler: () => window.location.reload() },
+		// Invoked when the navigator finishes loading
+		{
+			id: 'membrane.completeInitialization', handler: () => window.completeInitialization()
```

Comment:
So does vscode add all commands to the `window` object? I'm trying to figure out where `window.completeInitialization` is defined since it's not a built-in method, as far as I can tell.

Reply:
Good question! This is a convoluted way to communicate between our extension and the `ide.tsx` page in site.

 - Site adds a function to `window`
 - The extension can't call it because it runs in an iframe with a different `window`
 - So we create a command here to invoke it
 
I'm going to mark these with a `// HACK:` comment because, even though they don't show up in the command palette, they can still be invoked by the user with e.g. a keyboard shortcut.

At some point we'll figure out how to run our extension in the main frame and we'll be able to directly talk to `ide.tsx`

Reply:
cc @iamseeley for context ☝️